OWML.ModHelper.Menus/ModInputCombinationMenu.cs

```diff
@@ -0,0 +1,205 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using OWML.Common;
+using OWML.Common.Menus;
+using OWML.ModHelper.Events;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModInputCombinationMenu : ModPopupMenu, IModInputCombinationMenu
+    {
+        public event Action<string> OnConfirm;
+        public event Action OnCancel;
+
+        public List<IModInputCombinationElement> CombinationElements { get; private set; }
+
+        private List<Selectable> _selectables;
+
+        public string Combination
+        {
+            get
+            {
+                string result = "";
+                for (int i = 0; i < CombinationElements.Count; i++)
+                {
+                    while (i < CombinationElements.Count && CombinationElements[i].Title == "")
+                    {
+                        CombinationElements[i].DestroySelf();
+                    }
+                }
+                for (int i = 0; i < CombinationElements.Count; i++)
+                {
+                    if (CombinationElements[i].Title != "")
+                    {
+                        result += CombinationElements[i].Title;
+                        if (i < CombinationElements.Count - 1)
+                        {
+                            result += "/";
+                        }
+                    }
+                }
+                return result;
```

Comment:
Extract to method

Reply:
Why though?

Reply:
Public getters and setters that do this much work seems crazy to me, because when you use them, it isn't clear that all this extra work is happening. Using a `GetCombination` or `ComputeCombination` method would be better here.

Reply:
Yeah I agree with @Raicuparta

Reply:
Don't want magical code (code that does un-intuitive things)

Reply:
GenerateCombination and FillMenu now